### ISSUE TITLE
Extend filters to include NUTS regions

### ISSUE THREAD

**Issue by DavidRoy** [BiologicalRecordsCentre/iRecord#1059]
Required to support Bloomin' Algae notification/verification setup
Ideally down to NUTS3 level

**Comment by johnvanbreda** [comment]
Is this for UK (or British Isles?) only?

Note that NUTS has a new version for 2021, but it excludes the UK for obvious reasons. Therefore we will need to use the outdated 2016 NUTS 3 layer. Although this layer does have smaller subdivisions than Vice Counties (at least in more populated counties), is there a real benefit of using NUTS 3 vs Vice counties?

**Comment by johnvanbreda** [comment]
See https://gisco-services.ec.europa.eu/distribution/v2/nuts/download/ for downloads.

**Comment by LaurenceCarvalho** [comment]
We would initially like NUTS 2 and NUTS3 options set up for UK & Ireland (top priority) but ideally also for Belgium and the Netherlands (at least NUTS2).
What we specifically need in UK and Ireland are the local authority regions - as this is being set up for environmental health officers in LAs to receive notifications. I don't think the old vice counties really fit with these - I would have thought NUTS3 was more likely to be a fit?  
For Belgium and the Netherlands we can stick with NUTS2 for now.

I guess because the UK is our top priority region we should stick with the 2016NUTS3 - but it would probably be just as easy to provide you with polygons specifically for LAs in UK and Ireland?

Laurence

**Comment by DavidRoy** [comment]
I think it makes sense to use LAs since that is the structure for environmental health.
@LaurenceCarvalho - can you provide the LA layer and John can advise on work required to import the layer, index, extend the filter definition

**Comment by johnvanbreda** [comment]
Yes, I agree - better to use a valid LA layer than extract just the UK stuff from the out-of-date 2016 NUTS 3 layer. It will also make more sense if we use this layer elsewhere, e.g. in iRecord.

@LaurenceCarvalho are you able to provide a SHP file?

**Comment by LaurenceCarvalho** [comment]
Hi Laurence,

I’d made the attached for Gemma that was based on EA regions in England, then Wales, Scotland, NI and Ireland as polygons. This was for the app / mapping, but it sounds like this might be for a different purpose? Just want to check that the two things are different before I make these too. For Gemma’s use we had to spend a lot of time simplifying them so they’d work in the mapping as json, guessing that’s less of an issue here?

Thanks,
Phil



From: Carvalho, Laurence ***@***.***>
Sent: 18 May 2021 14:06
To: BiologicalRecordsCentre/iRecord ***@***.***>; BiologicalRecordsCentre/iRecord ***@***.***>; Taylor, Philip ***@***.***>
Cc: Mention ***@***.***>; Roy, David ***@***.***>; ***@***.***
Subject: RE: [BiologicalRecordsCentre/iRecord] Extend filters to include NUTS regions (#1059)

Hi Phil

We’re trying to set up the notification/verification service for Bloomin’Algae in iRecord at a local authority level in UK and Ireland.

Can you provide John Van Breda with the shape file for UK LA boundaries?

Can you get these for Ireland too? If not I will ask Deirdre as she is suggesting the WFD teams in Ireland based in LAs will be best placed to receive notifications.

Thanks

Laurence

**Comment by LaurenceCarvalho** [comment]
Hi Phil (and David and John)

I think these are different – I think the EA/NE polygons probably fit quite closely with NUTS 1 regions – but I think BRC already have shapefiles for these.

David/John – would it be possible to use these simplified shapefiles from Phil as the coarser area boundaries in iRecord equivalent to NUTS 1 regions?  Or do you prefer to stick with NUTS 1 so it is consistent across UK and European countries? These shapefiles are being used in the map of records on the Bloomin’Algae webpage

Phil - What we need are shapefiles for NUTS3 – this delineates UK local authorities (called unitary authorities in England and Wales, council areas in Scotland, districts in NI).

For the UK it looks like they’re here… https://data.gov.uk/dataset/f884db1c-38a1-4fa1-9d0e-3c4a0d80e2da/nuts-level-3-january-2018-full-clipped-boundaries-in-the-united-kingdom

For Ireland and Belgium they’re probably here… https://ec.europa.eu/eurostat/web/gisco/geodata/reference-data/administrative-units-statistical-units/nuts
I will ask Deirdre for  the Irish ones as they may be different for the Irish WFD monitoring regions which she says are based around LAs

Thanks

Laurence



From: Taylor, Philip ***@***.***>
Sent: 20 May 2021 09:39
To: Carvalho, Laurence ***@***.***>; BiologicalRecordsCentre/iRecord ***@***.***>; BiologicalRecordsCentre/iRecord ***@***.***>
Cc: Mention ***@***.***>; Roy, David ***@***.***>; ***@***.***
Subject: RE: [BiologicalRecordsCentre/iRecord] Extend filters to include NUTS regions (#1059)

Hi Laurence,

I’d made the attached for Gemma that was based on EA regions in England, then Wales, Scotland, NI and Ireland as polygons. This was for the app / mapping, but it sounds like this might be for a different purpose? Just want to check that the two things are different before I make these too. For Gemma’s use we had to spend a lot of time simplifying them so they’d work in the mapping as json, guessing that’s less of an issue here?

Thanks,
Phil



From: Carvalho, Laurence ***@***.******@***.***>>
Sent: 18 May 2021 14:06
To: BiologicalRecordsCentre/iRecord ***@***.******@***.***>>; BiologicalRecordsCentre/iRecord ***@***.******@***.***>>; Taylor, Philip ***@***.******@***.***>>
Cc: Mention ***@***.******@***.***>>; Roy, David ***@***.******@***.***>>; ***@***.******@***.***>
Subject: RE: [BiologicalRecordsCentre/iRecord] Extend filters to include NUTS regions (#1059)

Hi Phil

We’re trying to set up the notification/verification service for Bloomin’Algae in iRecord at a local authority level in UK and Ireland.

Can you provide John Van Breda with the shape file for UK LA boundaries?

Can you get these for Ireland too? If not I will ask Deirdre as she is suggesting the WFD teams in Ireland based in LAs will be best placed to receive notifications.

Thanks

Laurence

**Comment by LaurenceCarvalho** [comment]
Hi all,

The NUT3 regions can be downloaded at a number of scales. I’ve just had a look and can send through what looks like the best balance of detail / file size. Do you want them clipped to just UK, Ire & Belgium, or send them through for all of Europe?

Thanks,
Phil



From: Carvalho, Laurence ***@***.***>
Sent: 20 May 2021 10:03
To: Taylor, Philip ***@***.***>; BiologicalRecordsCentre/iRecord ***@***.***>; BiologicalRecordsCentre/iRecord ***@***.***>
Cc: Mention ***@***.***>; Roy, David ***@***.***>; ***@***.***
Subject: RE: [BiologicalRecordsCentre/iRecord] Extend filters to include NUTS regions (#1059)

Hi Phil (and David and John)

I think these are different – I think the EA/NE polygons probably fit quite closely with NUTS 1 regions – but I think BRC already have shapefiles for these.

David/John – would it be possible to use these simplified shapefiles from Phil as the coarser area boundaries in iRecord equivalent to NUTS 1 regions?  Or do you prefer to stick with NUTS 1 so it is consistent across UK and European countries? These shapefiles are being used in the map of records on the Bloomin’Algae webpage

Phil - What we need are shapefiles for NUTS3 – this delineates UK local authorities (called unitary authorities in England and Wales, council areas in Scotland, districts in NI).

For the UK it looks like they’re here… https://data.gov.uk/dataset/f884db1c-38a1-4fa1-9d0e-3c4a0d80e2da/nuts-level-3-january-2018-full-clipped-boundaries-in-the-united-kingdom

For Ireland and Belgium they’re probably here… https://ec.europa.eu/eurostat/web/gisco/geodata/reference-data/administrative-units-statistical-units/nuts
I will ask Deirdre for  the Irish ones as they may be different for the Irish WFD monitoring regions which she says are based around LAs

Thanks

Laurence



From: Taylor, Philip ***@***.******@***.***>>
Sent: 20 May 2021 09:39
To: Carvalho, Laurence ***@***.******@***.***>>; BiologicalRecordsCentre/iRecord ***@***.******@***.***>>; BiologicalRecordsCentre/iRecord ***@***.******@***.***>>
Cc: Mention ***@***.******@***.***>>; Roy, David ***@***.******@***.***>>; ***@***.******@***.***>
Subject: RE: [BiologicalRecordsCentre/iRecord] Extend filters to include NUTS regions (#1059)

Hi Laurence,

I’d made the attached for Gemma that was based on EA regions in England, then Wales, Scotland, NI and Ireland as polygons. This was for the app / mapping, but it sounds like this might be for a different purpose? Just want to check that the two things are different before I make these too. For Gemma’s use we had to spend a lot of time simplifying them so they’d work in the mapping as json, guessing that’s less of an issue here?

Thanks,
Phil



From: Carvalho, Laurence ***@***.******@***.***>>
Sent: 18 May 2021 14:06
To: BiologicalRecordsCentre/iRecord ***@***.******@***.***>>; BiologicalRecordsCentre/iRecord ***@***.******@***.***>>; Taylor, Philip ***@***.******@***.***>>
Cc: Mention ***@***.******@***.***>>; Roy, David ***@***.******@***.***>>; ***@***.******@***.***>
Subject: RE: [BiologicalRecordsCentre/iRecord] Extend filters to include NUTS regions (#1059)

Hi Phil

We’re trying to set up the notification/verification service for Bloomin’Algae in iRecord at a local authority level in UK and Ireland.

Can you provide John Van Breda with the shape file for UK LA boundaries?

Can you get these for Ireland too? If not I will ask Deirdre as she is suggesting the WFD teams in Ireland based in LAs will be best placed to receive notifications.

Thanks

Laurence

**Comment by ScienceAndMaps** [comment]
Sorry, some confusion.

I think what we actually want is Local Authorities for UK: https://geoportal.statistics.gov.uk/datasets/1d78d47c87df4212b79fe2323aae8e08_0

and NUTS3 for Ireland & Beligium? https://ec.europa.eu/eurostat/web/gisco/geodata/reference-data/administrative-units-statistical-units/nuts

I can sort them and send them through if that looks right.

**Comment by DavidRoy** [comment]
Thanks @ScienceAndMaps - agreed.  Thanks for the links.  John can access the shapefiles directly so no need for you to extract

@johnvanbreda 
Can you import the UK LA layer and NUTS level 3 into Indicia and enable them to be used in the filter tool within iRecord

**Comment by ScienceAndMaps** [comment]
That's great, thanks

**Comment by LaurenceCarvalho** [comment]
Thanks Phil/John
My impression from this website https://en.wikipedia.org/wiki/NUTS_statistical_regions_of_the_United_Kingdom was that NUTS3 (2016) was the same as UK local authorities - but happy for you to use the ONS shapefiles.  Maybe there's a GIS test for you Phil to see if they are comparable!

Laurence

**Comment by johnvanbreda** [comment]
@DavidRoy If the UK authorities are the same as the NUTS 3 2016 boundaries, then should we just import the entire NUTS 3 2016 layer and index it? Though that will of course have a performance impact on the background tasks.

Or should we clip the NUTS 3 2016 layer to just UK, Belgium & Ireland?

Or we could clip the NUTS3 2021 layer to Belgium & Ireland, then add the UK authorities separately?

**Comment by DavidRoy** [comment]
seems like next step is for @ScienceAndMaps to check if NUTS3 is equivalent to LAs for the UK

**Comment by ScienceAndMaps** [comment]
![NUTS3_LA](https://user-images.githubusercontent.com/26765140/119014432-e0541200-b98f-11eb-92af-70cb2f940c4d.gif)

They seem different. NUTS3 isn't as detailed. For example, it has 'East Lancashire' whereas LAs are for Pendle, Burnley etc. You might want to make a call on what to use, but LAs provide more regions.

**Comment by LaurenceCarvalho** [comment]
There is clearly an overlap. If we use NUTS3 then the public (and some LAs) would get notifications from neighbouring LAs - but maybe this is ok as it warns them of incidents nearby? Maybe more an issue for LA env health officers - but we haven't got them on board in England and maybe they wouldn't mind either?

Personally, I like the consistency of using NUTS3 across Europe.

**Comment by DavidRoy** [comment]
I'd also favour the larger/simpler NUTS3 boundaries
@johnvanbreda I suggest you include NUTS3 for the whole of Europe for simplicity.  We can deal with the performance of background tasks if and when they become an issue - unlikely initially as most data is from the UK

**Comment by ScienceAndMaps** [comment]
Sounds sensible to me.

**Comment by johnvanbreda** [comment]
@DavidRoy I've loaded NUTS3 as a layer. Should we re-index all data against it? Or will it just be for Bloomin' Algae at the moment?
Also, should it be exposed in the usual Explore tools (assuming the answer is to index all data)?

**Comment by LaurenceCarvalho** [comment]
Hi John/David
Do you have any idea of when this could be completed by?  We are hoping to have some training for verifiers in UK, Ireland, Belgium and Netherlands during June if possible
Even if just set up at country/NUTS2 level for starters that would be something we could use in training.

Do I need to setup a PO first? Also, John can you provide a quote for your time/work for the PO.

Thanks
Laurence

**Comment by kitenetter** [comment]
@johnvanbreda @DavidRoy will the NUTS boundaries (including non-UK) be available for setting up verification filters in the usual way?

Do we still need to make decisions about re-indexing, and about inclusion in the Explore tools? I don't see any problem with adding this option to the Explore tools but that is less of a priority than the alerts and verification filters.

The verification roles do apply only to the Bloomin Algae data, so we could confine re-indexing to that dataset (and thus not add to the Explore tools).

**Comment by johnvanbreda** [comment]
@kitenetter Yes, the NUTS layers are already available for the filters admin tool on iRecord. However, I have not re-indexed existing data for NUTS 3, so as it stands it will only work for data going forward (which includes Bloomin' Algae). 

I can re-index the whole lot at some point, but I don't think this is a high priority at this point?
